src/class-wpml-gutenberg-integration.php

```diff
@@ -138,37 +140,39 @@ public function string_translated(
 	 */
 	private function update_block_translations( $blocks, $string_translations, $lang ) {
 		foreach ( $blocks as &$block ) {
-			$block = $this->strings_in_blocks->update( $block, $string_translations, $lang );
-
-			if ( isset( $block['blockName'] ) && 'core/block' === $block['blockName'] ) {
-				$block['attrs']['ref'] = apply_filters( 'wpml_object_id', $block['attrs']['ref'], 'wp_block', true, $lang );
-			}
-			if ( isset( $block['innerBlocks'] ) ) {
-				$block['innerBlocks'] = $this->update_block_translations(
-					$block['innerBlocks'],
-					$string_translations,
-					$lang
-				);
+			if ( $block instanceof WP_Block_Parser_Block ) {
+				$block = $this->strings_in_blocks->update( $block, $string_translations, $lang );
+
+				if ( isset( $block->blockName ) && 'core/block' === $block->blockName ) {
+					$block->attrs['ref'] = apply_filters( 'wpml_object_id', $block->attrs['ref'], 'wp_block', true, $lang );
+				}
+				if ( isset( $block->innerBlocks ) ) {
+					$block->innerBlocks = $this->update_block_translations(
+						$block->innerBlocks,
+						$string_translations,
+						$lang
+					);
+				}
 			}
 		}
 
 		return $blocks;
 	}
 
 	/**
-	 * @param array $block
+	 * @param array|WP_Block_Parser_Block $block
```

Comment:
Can it still be an array @brucepearson?

Reply:
@strategio Yes, it can be an array. Gutenberg returns an array with `innerHTML` for anything that's not inside a block.